### PR TITLE
Role Management added

### PR DESCRIPTION
**Pull request by ma-gu**
Please review role management and merge into master when ready.

### CODE REVIEW

File: Infrastructure.Security/Services/Implementation/Passwords.cs [hunk 1]
```diff
@@ -8,9 +8,24 @@
 
 namespace Infrastructure.Security.Services.Implementation
 {
-    public class Passwords : IHandler<PasswordCreation>, IHandler<PasswordTokenCreation>, IHandler<ResetPassword>, IHandler<ChangePassword>
+    public class Passwords : IHandler<PasswordCreation>, IHandler<PasswordTokenCreation>, 
+		IHandler<ResetPassword>, IHandler<ChangePassword>, IHandler<ForgotPassword>
     {
-        public async Task<bool> HandleAsync(ResetPassword e)
+		public async Task<bool> HandleAsync(ForgotPassword e)
+		{
```
Comment (dmitrynogin) [inline Infrastructure.Security/Services/Implementation/Passwords.cs]: @ma-gu @johnnyoshika Proper names for events would be PasswordReset, PasswordForgotten, PasswordChange.

Comment (johnnyoshika) [inline Infrastructure.Security/Services/Implementation/Passwords.cs]: These events have been renamed:
https://github.com/bcjobs/infrastructure/commit/7f2e54630d1cd12c91191356088c4c2d795d4549

**Comment by johnnyoshika** [comment]
@ma-gu: I removed the ForgotPassword reset because it's not needed in our reset password scenario.  The password reset flow will go like this:

* User initiates a password reset by entering their email address.
  * UI layer will use the `PasswordTokenCreation` event to generate a reset password token and email it to the user.
* Once user receives the email, s/he will return to the site to reset her password using the `PasswordReset` event.

Also, please note that we can't use IdentityManager's email features, as email is not managed by our own classes outside of ASP.NET Identity, so this code won't work
```
var userManager = IdentityManagers.UserManager; 
var user = await userManager.FindByEmailAsync(e.Email);
```

**Comment by johnnyoshika** [comment]
Thank you @ma-gu  for this work and @dmitrynogin for your review.  I rebased then merged this into `master`.

**Pull request closed without merging**